app/views/chargeSummary.scala.html

```diff
@@ -130,6 +133,17 @@ <h2 class="heading-large">@messages("chargeSummary.chargeHistory.heading")</h2>
 						<td class="numeric">@{charge.totalAmount.toCurrency}</td>
 					</tr>
 				}
+			}
+
+				@if(paymentAllocationEnabled) {
+					@for(allocation <- paymentAllocations ; payment <- allocation.payments) {
+					<tr>
+						<td style="white-space: nowrap;">@LocalDate.parse(payment.date.get).toLongDateShort</td>
+						<td>@allocation.getPaymentAllocationTextInChargeSummary.map(messages(_))</td>
+						<td class="numeric">@payment.amount.map(_.abs.toCurrency)</td>
```

Comment:
Could you change it to payment.amount.get? By doing that we error out in this page rather than displaying blank value just like line 141 (or) it might be worth checking for nonEmpty values all the row values before displaying.